src/test/java/com/github/davidmoten/rx2/aws/SqsTest.java

```diff
@@ -292,63 +314,129 @@ public void testPollingReturnsAllAvailableMessagesAtEachScheduledCall() {
             final InOrder inorder = Mockito.inOrder(sqs, s3);
-            inorder.verify(s3, Mockito.times(1)).putObject(Mockito.anyString(), Mockito.anyString(), Mockito.any(),
-                    Mockito.any());
-            inorder.verify(sqs, Mockito.times(1)).sendMessage(Mockito.anyString(), Mockito.anyString());
-            inorder.verify(s3, Mockito.times(1)).deleteObject(Mockito.anyString(), Mockito.anyString());
+            inorder.verify(s3, Mockito.times(1)).putObject(Mockito.anyString(), Mockito.anyString(),
+                    Mockito.any(), Mockito.any());
+            inorder.verify(sqs, Mockito.times(1)).sendMessage(Mockito.anyString(),
+                    Mockito.anyString());
+            inorder.verify(s3, Mockito.times(1)).deleteObject(Mockito.anyString(),
+                    Mockito.anyString());
             inorder.verifyNoMoreInteractions();
         }
     }
-    
+
     @Test
     public void testCanUseNewAwsBuildersInFactoryMethods() {
-        Sqs.queueName("queue").sqsFactory(() -> AmazonSQSClientBuilder.standard().withRegion("ap-southeast-2").build());
+        Sqs.queueName("queue").sqsFactory(
+                () -> AmazonSQSClientBuilder.standard().withRegion("ap-southeast-2").build());
     }
-    // @SuppressWarnings("unused")
-    // public static void main(String[] args) {
-    // ClientConfiguration cc;
-    // if (false)
-    // cc = new
-    // ClientConfiguration().withProxyHost("proxy.amsa.gov.au").withProxyPort(8080);
-    // else
-    // cc = new ClientConfiguration();
-    // AWSCredentialsProvider credentials = new
-    // SystemPropertiesCredentialsProvider();
-    // Func0<AmazonSQSClient> sqs = () -> new AmazonSQSClient(credentials, cc)
-    // .withRegion(Region.getRegion(Regions.AP_SOUTHEAST_2));
-    // Func0<AmazonS3Client> s3 = () -> new AmazonS3Client(credentials, cc)
-    // .withRegion(Region.getRegion(Regions.AP_SOUTHEAST_2));
-    // String bucketName = "cts-gateway-requests";
-    // String queueName = bucketName;
-    // Sqs.queueName(queueName) //
-    // .sqsFactory(sqs) //
-    // .bucketName(bucketName) //
-    // .s3Factory(s3) //
-    // .messages() //
-    // .subscribeOn(Schedulers.io()) //
-    // .doOnNext(System.out::println) //
-    // .doOnNext(SqsMessage::deleteMessage) //
-    // .doOnError(e -> {
-    // e.printStackTrace();
-    // System.out.println(Thread.currentThread().getName());
-    // }) //
-    // .retryWhen(RetryWhen.delay(5, TimeUnit.SECONDS).build(), Schedulers.io())
-    // //
-    // .toBlocking().subscribe();
-    // }
 
+    @Test
+    public void testPollEvents() {
+        List<Message> list = new ArrayList<>();
+        List<String> events = new ArrayList<>();
+        List<Message> result = Sqs.messages(() -> list, () -> events.add("prePoll"), e -> events.add("postPoll"));
+        assertTrue(result == list);
+        assertEquals(Arrays.asList("prePoll", "postPoll"), events);
+    }
+    
+    @Test
```

Comment:
If the purpose of this test is to check an exception is thrown, shouldn't we use 
@Test(expected = RuntimeException.class)

Reply:
this pattern is used because another assertion happens after exception thrown